Library/Homebrew/extend/ARGV.rb

```diff
@@ -88,8 +88,9 @@ def next
   end
 
   def value(arg)
-    arg = find { |o| o =~ /--#{arg}=(.+)/ }
-    $1 if arg
+    return if arg.empty? || arg.index("-") || arg.index("=")
```

Comment:
I'd say malformed/invalid option names should raise an exception instead of just be ignored. Would make it easier for formula and `brew` authors to diagnose errors. And unlike options which simply aren't supported by a particular formula, it seems like there's never a case in which those option names would become valid, so it wouldn't rule out any functionality we use.